src/main/java/redis/clients/jedis/JedisSlotBasedConnectionHandler.java

```diff
@@ -44,9 +43,12 @@ public Jedis getConnection() {
         if (result.equalsIgnoreCase("pong")) return jedis;
 
         pool.returnBrokenResource(jedis);
-      } catch (JedisConnectionException ex) {
+      } catch (JedisException ex) {
         if (jedis != null) {
-          pool.returnBrokenResource(jedis);
+          try {
```

Comment:
> I am not sure whether we can get such exception here, but throwing it further up seems inappropriate, so I added the try-catch block.

@serge-groshev I prefer not to through it up because otherwise we could be hiding bugs that can be difficult to trace. defensive programming usually is not a good practice. That `try` block __shoudln't__ be necessary there, if it throws an exception is because something it's wrong and something else needs to be done. 

Also, the `returnBrokenResource` exception in #1168 comes from a different place, not from the `getConnection` method, so that's another smell that makes me think we shouldn't do anything here. Just to make sure, can you remove that `try-catch` block and run your tests again to see what we get?


> please note that if ping() does not throw, but returns anything else than "pong", the same method returnBrokenResource() is invoked.

Ping shoudn't return anything but "pong" or an exception. If it does, then it's a bug in the code which should be treated as such. Having said that, I guess we can change the whole codeblock to something like this:

```java
    for (JedisPool pool : pools) {
      Jedis jedis = null;
      try {
        jedis = pool.getResource();

        if (jedis == null) {
          continue;
        }

        String result = jedis.ping();

        if (result.equalsIgnoreCase("pong")) return jedis;

      } finally {
        if (jedis != null) {
          jedis.close();
        }
      }
    }
```
Since we'll always return the connection to the pool, we can just remove the exception handling and add a `finally` clause.

> And if ping() returns null, we get uncaught NPE there.

Again, defensive programming here. In theory should not happen, and if it does, it should be something we need to fix somewhere else, but not here.